app/src/main/java/org/glucosio/android/analytics/GoogleAnalytics.java

```diff
@@ -20,6 +23,9 @@ public void init(@NonNull final Context context) {
         mTracker = analytics.newTracker(BuildConfig.GOOGLE_ANALYTICS_TRACKER);
         mTracker.enableAdvertisingIdCollection(true);
 
+        SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(context);
+        enabled = sharedPref.getBoolean("pref_analytics_opt_in", false);
+
         if (BuildConfig.DEBUG) {
             com.google.android.gms.analytics.GoogleAnalytics.getInstance(context).setAppOptOut(true);
```

Comment:
This is what we do for debug configuration. We can call setAppOptOut here if the opt out preference is setted.